db/db_range_del_test.cc

```diff
@@ -1427,6 +1427,49 @@ TEST_F(DBRangeDelTest, SnapshotPreventsDroppedKeys) {
   db_->ReleaseSnapshot(snapshot);
 }
 
+TEST_F(DBRangeDelTest, SnapshotPreventsDroppedKeysInIMMemTables) {
+  const int kFileBytes = 1 << 20;
+
+  Options options = CurrentOptions();
+  options.compression = kNoCompression;
+  options.disable_auto_compactions = true;
+  options.target_file_size_base = kFileBytes;
+  Reopen(options);
+
+  // block flush thread -> pin immtables in memory
+  std::shared_ptr<std::atomic<bool>> verified(new std::atomic<bool>(false));
+  rocksdb::SyncPoint::GetInstance()->SetCallBack(
+      "DBImpl::BGWorkFlush", [&, verified](void*) {
+        while (!(*verified)) {
+          env_->SleepForMicroseconds(1000);
```

Comment:
We try to avoid sleep in tests now. You can use sync point instead:

- Replace the `verified->store(true)` with a `TEST_SYNC_POINT("DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified")`.
- Use `rocksdb::SyncPoint::GetInstance()->LoadDependency()` to make "DBImpl::BGWorkFlush" depend on the sync point "DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified". That will ensure the flush is blocked until this test's verification finishes.

Reply:
> We try to avoid sleep in tests now. You can use sync point instead:
> 
> * Replace the `verified->store(true)` with a `TEST_SYNC_POINT("DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified")`.
> * Use `rocksdb::SyncPoint::GetInstance()->LoadDependency()` to make "DBImpl::BGWorkFlush" depend on the sync point "DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified". That will ensure the flush is blocked until this test's verification finishes.

done ~

Reply:
> We try to avoid sleep in tests now. You can use sync point instead:
> 
> * Replace the `verified->store(true)` with a `TEST_SYNC_POINT("DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified")`.
> * Use `rocksdb::SyncPoint::GetInstance()->LoadDependency()` to make "DBImpl::BGWorkFlush" depend on the sync point "DBRangeDelTest::SnapshotPreventsDroppedKeysInIMMemTables:Verified". That will ensure the flush is blocked until this test's verification finishes.

done ~